### PR TITLE
fix: Add Index out of bound exception handling for PropertiesFile.save()

### PR DESCRIPTION
**Pull request by bohan-amplitude**
### Summary

Expection catch and log for PropertiesFile.save()
Change more general Exception catch for PropertiesFile.load()
Unit test for added function

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: id/src/main/java/com/amplitude/id/utilities/PropertiesFile.kt [hunk 1]
```diff
@@ -32,8 +32,12 @@ class PropertiesFile(directory: File, key: String, prefix: String, logger: Logge
     }
 
     private fun save() {
-        FileOutputStream(propertiesFile).use {
-            underlyingProperties.store(it, null)
+        try {
+            FileOutputStream(propertiesFile).use {
+                underlyingProperties.store(it, null)
+            }
+        } catch (e: IndexOutOfBoundsException) {
```
Comment (justin-fiedler) [inline id/src/main/java/com/amplitude/id/utilities/PropertiesFile.kt:39]: Is this the only IO error that can happen? We might want to catch all IO errors

Comment (bohan-amplitude) [reply id/src/main/java/com/amplitude/id/utilities/PropertiesFile.kt]: Changed to more general exception catch

**Review by justin-fiedler — APPROVED** [review]
This should workaround the issue. Were you able to locate the root cause? Is this from lack of file access or is it the Properties have become corrupted.

**Review by yuhao900914 — APPROVED** [review]

**Comment by bohan-amplitude** [comment]
> This should workaround the issue. Were you able to locate the root cause? Is this from lack of file access or is it the Properties have become corrupted?

I'll ask if customer can give us some log info that Marvin previous added. It sounds like a random access issue that caused by multithreading

**Comment by ondrej** [comment]
Will the logs you and Marvin added provide more info about the root cause? It looks like we're logging the file path and the stack trace, but the problem appears to be with the values in the properties map.

Do we blanket catch Exception normally in our SDKs? It's normally an anti-pattern since it can potentially hide serious problems that should actually escalate instead of leaving the app in an uncertain state.

**Pull request merged**